bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,326 @@
+The arguments passed to the bootable JAR are composed of bootable specific arguments and server arguments.
+
+Calling  java <JVM and system properties> -jar <bootable JAR name>.jar --help dumps the available arguments.
+
+==== Log manager
+
+The bootable JAR depends on JBoss log manager. Logging is configured thanks to the logging subsystem.
+
+==== Shutting down the Bootable JAR
+
+The bootable JAR process can be shutdown in the following ways:
+
+* Signal handling
+* :shutdown management operation with the following limitations:
+** The restart option is not supported and would be ignored by the server. The process will exit in all cases.
+
+
+==== Read Only standalone.xml
+
+The changes made to the management model are not reflected in the standalone.xml. 
+The server is internally started with --read-only-server-config=standalone.xml
+
+==== Arguments handling
+
+Arguments specific to bootable JAR:
+
+|===
+|Option |Description
+|--help
+|Display help then exit
+
+|--deployment=<path to WAR/JAR/EAR file or exploded deployment directory>
+|Application to install in the hollow JAR. Adding a deployment to an bootable JAR already containing a deployment is invalid.
+
+|--install-dir=<path to directory to install server in>
+|By default a new TEMP directory is created.
+|===
+
+Server arguments:
+
+|===
+|Option |Description
+|-b[interface]=<value>
+|Set system property jboss.bind.address.<interface> to the given value
+
+|-b=<value>
+|Set system property jboss.bind.address to the given value
+
+|-D<name>[=<value>] 
+|Set a system property. The system properties are set by the server. They are not set by the bootable JAR JVM.
+
+|-u=<value>
+|Set system property jboss.default.multicast.address to the given value.
+
+|--version
+|Print version and exit.
+
+|-S<name>[=value]
+|Set a security property
+
+|--properties=<url>
+|Load system properties for the given url
+|===
+
+==== Developer experience
+
+Having to rebuild a server and package a bootable JAR for each code change is not a valid approach. 
+The plugin should offer a development mode allowing to make development using bootable JAR an efficient task. 
+We are defining here a workflow that leverages the hollow JAR packaging and server deployment scanner capabilities.
+
+===== Dev mode
+
+* A dev server is an hollow server scanning the directory target/deployments.
+* A dev app is the primary artifact copied to the target/deployments dir.
+
+The plugin attempts to force the provisioning of the deployment scanner in dev mode:
+
+* If deployment-scanner is excluded, remove it from exclusion (Warn).
+* Add the deployment-scanner layer to the set of layers (Info)
+* If a provisioning.xml file is set (and no layers override it), a warning is displayed advertising that we can’t enforce the presence of the scanner. (Warn)
+* If the deployment-scanner is not in the config, the CLI operation will abort the plugin execution. (Error)
```

Comment:
I'd suggest to s/`. (Error)`/` and log an error message.`/ here.

Reply:
Done